### PR TITLE
Algorithm

### PR DESCRIPTION
**Pull request by Stapa1**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by tianyizheng02** [comment]
There's currently an implementation of Strassen's matrix multiplication algorithm in [divide_and_conquer/strassen_matrix_multiplication.py](https://github.com/TheAlgorithms/Python/blob/master/divide_and_conquer/strassen_matrix_multiplication.py). Is this implementation different from that one?

**Comment by Vishnuvardhan-STA** [comment]
Yes it's different

**Comment by Stapa1** [comment]
> There's currently an implementation of Strassen's matrix multiplication algorithm in [divide_and_conquer/strassen_matrix_multiplication.py](https://github.com/TheAlgorithms/Python/blob/master/divide_and_conquer/strassen_matrix_multiplication.py). Is this implementation different from that one?
yes this implementation different from that one.

File: backtracking/Magnet_Puzzle.py3 [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Write Python3 code here
```
Comment (tianyizheng02) [inline backtracking/Magnet_Puzzle.py3:1]: Plagiarized from [GeeksforGeeks](https://www.geeksforgeeks.org/magnet-puzzle-backtracking-9/). Code is identical down to the comments, only differing in some whitespace.

@cclauss Thoughts?

**Comment by cclauss** [comment]
## [x]  This pull request is all my own work -- I have not plagiarized.

**Pull request closed without merging**